android/intltest/java/com/facebook/hermes/test/HermesIntlNumberFormatTest.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import android.content.res.AssetManager;
+import android.icu.util.ULocale;
+import android.test.InstrumentationTestCase;
+import android.text.TextUtils;
+import android.util.Log;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+
+import com.facebook.hermes.test.JSRuntime;
+
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+// Run "./gradlew :intltest:preparetest262" from the root to download and copy the test files to the APK assets.
+public class HermesIntlNumberFormatTest extends HermesIntlTest262Base {
+
+    private static final String LOG_TAG = "HermesIntlNumberFormatTest";
+
+    public void testIntlNumberFormat() throws IOException {
+
+        String basePath = "test262-main/test/intl402/NumberFormat";
+
+        Set<String> whiteList = new HashSet<>();
+
+        Set<String> deviations = new HashSet<>(Arrays.asList(
+                "dft-currency-mnfd-range-check-mxfd.js", // This test is not correct based on spec. With ths currency in test, the default minfractiondigits is 2, and the maxfractiondigita cannot be set to 1. Both Firefox and Chrome also fails the test.
+                "constructor-order.js",  // When strictly following spec, the currency checks comes before unit check and the test will throw RangeError on currency validation before reaching the cdoe which throws TypeError on seeing undefined unit. But, we have a part of option validation in C++ code which throws all TypeErrors, which results in the TypeError getting thrown.
+                "currency-digits.js", // Didn't get correct minimumFractionDigts for currency AFN. Expected SameValue(«0», «2») to be true
+                "constructor-unit.js" // com.facebook.hermes.intl.JSRangeErrorException: Unknown unit: acre-per-acre .. We support only units directly known to https://developer.android.com/reference/android/icu/util/MeasureUnit .. MeasureFormat.format requires an instance of MeasureUnit.
+        ));
+
+        Set<String> testIssues = new HashSet<>(Arrays.asList(
+                "proto-from-ctor-realm.js", // Property '$262' doesn't exist
```

Comment:
defined in harness/detachArrayBuffer-host-detachArrayBuffer.js

Reply:
That one doesn't seem to be the one.. 

But https://github.com/tc39/test262/blob/20e31011058783d9bdeb4c34a8aacbb632355e2e/implementation-contributed/v8/test262/harness-adapt.js#L107 looks more like the one, which adds $262 to global with RealmOperators .. 

Do we have something like it for Hermes?

Reply:
It looks like we do, it's here: https://github.com/facebook/hermes/blob/ad7229435fdd6ab45beff44731a955067c1c072a/utils/testsuite/testsuite.py#L130-L137

Unfortunately, it's inlined in the py file right now.  We should be able to pull it into a standalone js file you can use.

Reply:
Let me try it today.